test/qunit/tests/list-of-pairs-collection-creator.js

```diff
@@ -16,6 +16,31 @@ define([
 
     module( "Galaxy client app tests" );
 
+    test( "Collection creation", function() {
+        var pcc = new PCC({
+                datasets    : DATA._1,
+                historyId   : 'fakeHistoryId'
+            }),
+            server = sinon.fakeServer.create();
+
+        var requestJSON;
+        server.respondWith( 'POST', '/api/histories/fakeHistoryId/contents/dataset_collections', function( request ){
+            requestJSON = JSON.parse( request.requestBody );
+            request.respond(
+                200,
+                { "Content-Type": "application/json" },
+                JSON.stringify({
+                    fakeResponse: 'yes'
+                })
+            );
+        });
+
+        //console.debug( 'requestBody:', JSON.stringify( requestJSON, null, '  ' ) );
+        pcc.createList( 'Heres a collection' );
+        server.respond();
+        deepEqual( requestJSON, DATA._1requestJSON );
+    });
+
```

Comment:
It does fail the test *every time* when using: `grunt --test=list-of-pairs-collection-creator.html`

The ids are generated in _ensureIds in list-of-pairs-collection-creator. Generally it's never hit because, like the test data here, the datasets used already have ids. It was meant for datasets that aren't even *in* galaxy yet - i.e. they were planned so we could generate collections from the upload form.

I'm not entirely sure why (or if) that's what's causing the error here. I'll check it out - thanks